lib/routes/infoq/presentations.ts

```diff
@@ -0,0 +1,203 @@
+                images: image
+                    ? [
+                          {
+                              src: image,
+                              alt: title,
+                          },
+                      ]
+                    : undefined,
+                intro: item.find('p.card__excerpt').text(),
+            });
+            const link = new URL(a.prop('href'), rootUrl).href;
+            const guid = `infoq-${item.prop('data-path').replace(/^\//, '')}`;
+            const length = item.find('div.card__length').text() || undefined;
+
+            return {
+                title,
+                description,
+                pubDate: parseDate(item.find('span.card__date span').text().trim()),
+                link,
+                category: item
+                    .find('div.card__topics')
+                    .toArray()
+                    .map((c) => $(c).text().trim()),
+                author: item
+                    .find('div.card__authors a')
+                    .toArray()
+                    .map((a) => $(a).text().trim())
+                    .join('/'),
+                guid,
+                id: guid,
+                content: {
+                    html: description,
+                    text: item.find('p.card__excerpt').text(),
+                },
+                image,
+                banner: image,
+                language,
+                enclosure_url: length ? link : undefined,
+                enclosure_type: length ? 'video/mp4' : undefined,
+                enclosure_title: title,
+                itunes_duration: length,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const { data: detailResponse } = await got(item.link);
+
+                const $$ = load(detailResponse);
+
+                $$('div.player').prevAll().remove();
+                $$('div.event__list-box').remove();
+
+                const length = $$('div.player__actions span').text() || undefined;
+
+                const script = $$('script[type="text/javascript"]').text();
+                const videoSrc = script.match(/P\.s\s=\s'(.*?)';/)?.[1] ?? undefined;
+                const poster = script.match(/P\.c\(.*?isWideScreen,\s'(.*?)',\s/)?.[1] ?? undefined;
+                const topicsStr = script.match(/var\stopicsInPage\s=\sJSON\.parse\('(.*?)'\);/)?.[1]?.replace(/\\/g, '') ?? undefined;
+
+                if (videoSrc) {
+                    $$('div.player').replaceWith(
+                        art(path.join(__dirname, 'templates/description.art'), {
+                            videos: [
+                                {
+                                    src: videoSrc,
+                                    poster,
+                                    type: videoSrc ? `video/${videoSrc.split(/\./).pop()}` : undefined,
```

Comment:
Condition 'videoSrc' is always true at this point because it is already checked at line 97

Reply:
👌